store/src/java/com/zimbra/cs/mime/MimeHandler.java

```diff
@@ -177,11 +177,14 @@ public void setSize(long value) {
      */
     public final String getContent() throws MimeHandlerException {
         if (!DebugConfig.disableMimePartExtraction) {
-            String toRet = getContentImpl();
-            if (toRet == null)
-                return "";
-            else
-                return toRet;
+            String toRet = "";
+            try {
+                toRet = getContentImpl();
+            } catch (OutOfMemoryError e) {
+                // java.lang.OutOfMemoryError: Java heap space handled here
+                ZimbraLog.index.warn("File %s is corrupted causing OOM", filename);
```

Comment:
This handles the error but the usage of heap for the mime wil still continue right?

Reply:
We have no control there. but i see 
now its exiting from reindexing that account.Earlier it was giving OOM and sometime crashing mailbox.

Reply:
can we apply mime content size check with LDAP attrs we have along with the exception handling to avoid the exception

Reply:
Are you indicating about mime content size check "convertd_max_response_size"
In that case we can make following change from where this method is called.
` if ((isMainBody && (!handler.runsExternally() || indexAttachments)) ||
                            (indexAttachments && !DebugConfig.disableIndexingAttachmentsTogether && handler.getSize() < DebugConfig.convertdMaxResponseSize)) {
                    toRet = handler.getContent();
                }`

If not then which LDAP attr you are talking about?

Reply:
I guess you were checking `mailboxd_java_heap_size` for heap size, can we use this to check the attachment size before loading to stop the error